### ISSUE TITLE
Update grpcio dependency version

### ISSUE THREAD

**Issue by simplylizz** [dagster-io/dagster#10588]
### What's the use case?

https://github.com/dagster-io/dagster/blob/2d2011d1f54b82d145afef6752833e3bdd0a0f72/python_modules/dagster/setup.py#L70 - currently you have a restriction that grpcio version must be `<1.48.1`. As far as I understand the only reason for this constraint is that there are some issues with this particular release. But the most recent version of grpcio is `1.50.0`.

Requesting this because it looks like there is no binary wheel for python 3.11 & older versions of grpcio and thus it requires building from sources, what is a quite time consuming process and requires to install a lot of non-runtime dependencies (my build was still failing after ~5 mins of compilation), while `grpcio==1.50.0` installs just within a second.

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by joshuataylor** [comment]
Maybe as a workaround there could be documentation around how to install from whl sources? I know this is especially a PITA for Apple Silicon (M1/M2) users where there aren't wheels for everything. Python 3.11 helps this as many packages needed to upgrade to support CI to support 3.11, so they included wheel files for Apple anyway.

(grpcio still doesn't include whl's for 3.11, but I've built wheels for 3.11 and 1.47.2 here: https://github.com/joshuataylor/grpcio-mac-arm-build/actions/runs/3551376660 , download the artifact). Based off this: https://github.com/pietrodn/grpcio-mac-arm-build

**Comment by joshuataylor** [comment]
Here's what we ended up doing for M1:

In Poetry we have added grpcio with the version that dagster uses. We have vendored the file (it's 3.7mb, so not too bad..), I'm not sure if you can do url='' (this would be great, you could just throw this on S3. If someone does know if this works, let me know and I'd be happy to mirror it).

```
grpcio = [
    {markers="sys_platform == 'darwin' and platform_machine=='arm64'", path='vendored/grpcio-1.47.2-cp310-cp310-macosx_11_0_arm64.whl'},
    {version="^1.47.2", markers="platform_machine=='x86_64'", source="pypi"}
]
```

**Comment by davetapley** [comment]
Can I suggest setting a `python_requires < 3.11`  until this is resolved?
See also:
- https://github.com/dagster-io/dagster/issues/6987

**Comment by combiz** [comment]
> Hey @simplylizz - I totally hear the frustration here. Unfortunately the upstream issue here with hanging grpc clients is still happening on 1.50.0: [grpc/grpc#30843](https://github.com/grpc/grpc/issues/30843)

The latest grpcio==1.51.1 includes pre-built whl's for Python 3.11 (https://pypi.org/project/grpcio/#files). Would it now be possible to update the dagster deps ("grpcio>=1.32.0,<1.48.1") to avoid downgrading Python?

**Comment by gibsondan** [comment]
the main blocker for removing the pin is the upstream hanging issue with grpcio, which I believe is not Python version specific although we can verify that.

**Comment by combiz** [comment]
I have a Docker image with Python 3.11 and grpcio==1.51, and grpcio installs normally without any wheel building or timeout hangs. The pip install of dagster downgrades grpcio and this results in a long wheel rebuild and timeout hang.

**Comment by gibsondan** [comment]
We did some more testing and are only able to reproduce the hanging issues on 3.9 and earlier, so we've limited the pin to just `python_version < 3.10` in the release that's going out tomorrow.

**Comment by simplylizz** [comment]
So I guess this could be closed? At least my issues is solved. Thanks for reviewing it (also, feel free to reopen if you think it still makes sense).

**Comment by gibsondan** [comment]
It gives me no pleasure to report that upon further inspection we are seeing the same hangs in grpcio on Python 3.10 after all :/ Very likely due to some combination of https://github.com/grpc/grpc/issues/31885 as well as the aforementioned upstream grpc issue. We will likely need to bring back the pin (although you are welcome to override the pin and install a later version of grpcio if you'd like to risk it)

**Comment by kkom** [comment]
Hello! I'd like to share a semi-related piece of information here. Recently grpcio 1.51.3 was released, which finally ships with ARM wheels for Python: https://github.com/grpc/grpc/releases/tag/v1.51.3 (see https://github.com/grpc/grpc/issues/31737#issuecomment-1377539815 too)

We've learned how to vendor the ARM build from the https://github.com/pietrodn/grpcio-mac-arm-build prior to that release, but obviously not having to do this is much much nicer (we happily simplified the dependencies of our main project after 1.51.3 was finally released).

Am I right in understanding that the hanging issue with grpcio is probably still happening with >=1.51.3? Either way, perhaps it's worth checking this again – as the ARM builds situation is a big quality-of-life improvement for developers using recent Mac machines.

**Comment by gibsondan** [comment]
Hi Konrad - grpcio shipped a 1.47.5 release last week with ARM wheels as
well, so we are out of the woods there for the time being. We’re still
working closely with the gRPC team on fixing the upstream issues so that we
can remove the pin - hopefully more to share there soon.

**Comment by kkom** [comment]
It's great to hear that you're working with the gRPC team. After commenting I did see your recent activity on https://github.com/grpc/grpc/issues/31885 – really appreciate it and I'm keeping my fingers crossed!

I saw the [1.47.5](https://pypi.org/project/grpcio/1.47.5/#files) Dagster-compatible release as well. It seems to ship just Python 3.10, and not 3.11, wheels – but that's not the end of the world and definitely sufficient for testing Dagster.

However, using `python = "^3.10"` and `grpcio = "1.47.5"`, while installation succeeds without any issues, unfortunately I do get a `malloc` issue running the Dagster service (I'm on an M2 Pro machine):

```
2023-03-04 19:33:25 +0000 - dagster - INFO - Launching Dagster services...
python(64011,0x1f6d68140) malloc: Corruption of free object 0x14be04420: msizes 8/7 disagree
python(64011,0x1f6d68140) malloc: *** set a breakpoint in malloc_error_break to debug
python(64010,0x1f6d68140) malloc: Corruption of free object 0x15b64caf0: msizes 8/7 disagree
python(64010,0x1f6d68140) malloc: *** set a breakpoint in malloc_error_break to debug
2023-03-04 19:33:30 +0000 - dagster - INFO - Shutting down Dagster services...
2023-03-04 19:33:30 +0000 - dagster - INFO - Dagster services shut down.
```

What would be the most helpful next steps from my end? Would you like me to file a new issue in the `dagster-io/dagster` repository with these details, continue discussing here, or simply... wait until `grpcio>1.51.3` works better?

**Comment by gibsondan** [comment]
Is that happening in VSCode's terminal? Another user reported that error message and this seemed to fix it (unsetting the `MallocNanoZone` environment variable): https://discuss.dagster.io/t/8139439/hello-i-am-trying-to-deploy-a-github-project-to-dagster-clou#aa1bdd97-d5dc-4673-985b-73fe442ee0d9

**Comment by kkom** [comment]
Thank you! Yes it was that. The suggested VSCode settings fix works:

```
"terminal.integrated.env.osx": {
        "MallocNanoZone": "1",
  }
```

Happy to report that everything works now! Good luck solving the upstream issue, so that we can get on the latest grpcio.

**Comment by emirkmo** [comment]
Is it possible to link to a whl of grpcio for py3.11 that is compatible with dagster? Really annoying to delay the update to 3.11 of the whole stack because of this.

Not for ARM M1/M2 but regular X86 at least on debian based linux distros.

Trying to even build it is failing for me on latest debian with py3.11 not through vscode or anything, but in a conda venv or a poetry venv inside regular default terminal, with zsh. 

Forgive me if I am missing something basic.

**Comment by dstalzjohn** [comment]
Currently, for me it seems not possible to install both tensorflow==2.12 and dagster==1.3.7 in the same poetry environment because dagster and tensorflow require different grpcio versions like mentioned above. 
Do you have any idea when it will be possible to use grpcio versions greater 1.48.2 with dagster?

**Comment by simplylizz** [comment]
JFYI: it's possible now if you're using a newer Python version. See the most recent versions pin.

**Comment by dstalzjohn** [comment]
Thank you. I tried 3.10 prviously and that didn't work. With 3.11 I got a different error regarding tensorflow-macos on a M1. I think I have to wait till it is also possible for lower python versions.

**Comment by mgierada** [comment]
Can we close this issue? Seems it is solved in the [1.3.12](https://github.com/dagster-io/dagster/releases/tag/1.3.12) release
